django/api/services/calculate_rebate.py

```diff
@@ -0,0 +1,102 @@
+from ..models.go_electric_rebate_application import GoElectricRebateApplication
+from ..models.household_member import HouseholdMember
+from ..settings import INCOME_REBATES
+from enum import Enum
+
+
+class RebateType(Enum):
+    A = "A"
+    B = "B"
+    C = "C"
+    D = "Not Approved"
+
+
+def check_individual(primary_income):
+    if primary_income is None:
+        return RebateType.D.value
+    primary_income = int(primary_income)
+    if primary_income > INCOME_REBATES.get(RebateType.C.value).get("individual_income"):
+        return RebateType.D.value
+    elif primary_income <= INCOME_REBATES.get(RebateType.A.value).get(
+        "individual_income"
+    ):
+        return RebateType.A.value
+    elif primary_income <= INCOME_REBATES.get(RebateType.B.value).get(
+        "individual_income"
+    ):
+        return RebateType.B.value
+    elif primary_income <= INCOME_REBATES.get(RebateType.C.value).get(
+        "individual_income"
+    ):
+        return RebateType.C.value
+
+
+def check_household(primary_income, secondary_income):
+    if (primary_income is None) | (secondary_income is None):
+        return RebateType.D.value
+    household_income = int(primary_income) + int(secondary_income)
+    if household_income > INCOME_REBATES.get(RebateType.C.value).get(
+        "household_income"
+    ):
+        return RebateType.D.value
+    if household_income <= INCOME_REBATES.get(RebateType.A.value).get(
+        "household_income"
+    ):
+        return RebateType.A.value
+    elif household_income <= INCOME_REBATES.get(RebateType.B.value).get(
+        "household_income"
+    ):
+        return RebateType.B.value
+    elif household_income <= INCOME_REBATES.get(RebateType.C.value).get(
+        "household_income"
+    ):
+        return RebateType.C.value
+
+
+def get_final_rebate(individual_rebate, household_rebate):
+    if household_rebate == RebateType.A.value:
+        return INCOME_REBATES.get(RebateType.A.value).get("rebate")
+    if (
+        individual_rebate == RebateType.B.value
+        or household_rebate == RebateType.B.value
+    ):
+        return INCOME_REBATES.get(RebateType.B.value).get("rebate")
+    if (
+        individual_rebate == RebateType.C.value
+        or household_rebate == RebateType.C.value
+    ):
+        return INCOME_REBATES.get(RebateType.C.value).get("rebate")
+    if (
+        household_rebate == RebateType.D.value
+        and individual_rebate == RebateType.D.value
+    ):
+        return RebateType.D.value
+
+
+def calculate_rebate_amount(cra_response):
+    application_id = list(cra_response.keys())[0]
+    application = cra_response.get(application_id)
+    filtered_applications = GoElectricRebateApplication.objects.select_related(
+        "householdmember"
+    ).get(id__in=[application_id])
```

Comment:
ex: this id__in would be 

```["B5t92XeH7NnFUwxc", "9uXLvNQS5vkKnscD", "ctW8gU57YX4xfQ9o"]```

Reply:
You're probably going to want to break this fetch code up into a higher function which then calls and checks individual applicaitons.

That's sort of what I was suggesting with:

```
filtered_applications = GoElectricRebateApplication.objects.select_related(
        "householdmember"
    ).get(id__in=["B5t92XeH7NnFUwxc", "9uXLvNQS5vkKnscD", "ctW8gU57YX4xfQ9o")

# iterate over filtered_applications

for application in filtered_applications:
    some_new_function(application, cra_response.get(application.id))

```

Reply:
You don't really need to use enumerate, you can use `application.sin`, `hm = application.householdmember` to get the related obj and `hm.sin`